doc/specs/vulkan/chapters/memory.txt

```diff
@@ -775,29 +772,29 @@ same memory object.
 When memory is allocated, its contents are undefined.
 
 There is an implementation-dependent maximum number of memory allocations
-which can: be simultaneously created on a device.
+that can: be simultaneously created on a device.
 This is specified by the
 <<features-limits-maxMemoryAllocationCount,pname:maxMemoryAllocationCount>>
 member of the sname:VkPhysicalDeviceLimits structure.
 If pname:maxMemoryAllocationCount is exceeded, fname:vkAllocateMemory will
 return ename:VK_ERROR_TOO_MANY_OBJECTS.
 
-[NOTE]
-.Note
-====
 Some platforms may: have a limit on the maximum size of a single allocation.
 For example, certain systems may: fail to create allocations with a size
 greater than or equal to 4GB.
 Such a limit is implementation-dependent, and if such a failure occurs then
-the error ename:VK_ERROR_OUT_OF_DEVICE_MEMORY should: be returned.
-====
+the error ename:VK_ERROR_OUT_OF_DEVICE_MEMORY must: be returned.
 
 .Valid Usage
 ****
-  * [[VUID-vkAllocateMemory-device-00636]]
-    The number of currently valid memory objects, allocated from
-    pname:device, must: be less than
-    sname:VkPhysicalDeviceLimits::pname:maxMemoryAllocationCount
+  * pname:pAllocateInfo::pname:allocationSize must: be less than or equal to
+    sname:VkPhysicalDeviceMemoryProperties::pname:memoryHeaps[pname:pAllocateInfo::pname:memoryTypeIndex].pname:size
+    as returned by flink:vkGetPhysicalDeviceMemoryProperties for physical
+    device parent of pname:device
```

Comment:
I appreciate the extra clarity by identifying the parent, but in practice we don't do this anywhere else in the spec so it feels out of place. I'm not too unhappy with keeping it if you really want it though, but then I'd suggest something like:

```
...for the slink:VkPhysicalDevice which pname:device was created from.
```

Reply:
You don't? E.g. `VkBufferCreateInfo`:

>If sharingMode is VK_SHARING_MODE_CONCURRENT, each element of pQueueFamilyIndices must be unique and must be less than pQueueFamilyPropertyCount returned by either vkGetPhysicalDeviceQueueFamilyProperties or vkGetPhysicalDeviceQueueFamilyProperties2KHR for the physicalDevice that was used to create device

Should probably copy-past it thought to match...

Reply:
Not *usually*. That one's a bit weird :P

Change the wording though and I'm happy :)